test/cql-pytest/test_tools.py

```diff
@@ -0,0 +1,156 @@
+import json
+import nodetool
+import os
+import pytest
+import subprocess
+import util
+
+
+def simple_no_clustering_table(cql, keyspace):
+    table = util.unique_name()
+    schema = f"CREATE TABLE {keyspace}.{table} (pk int PRIMARY KEY , v int)"
+
+    cql.execute(schema)
+
+    for pk in range(0, 10):
+        cql.execute(f"INSERT INTO {keyspace}.{table} (pk, v) VALUES ({pk}, 0)")
+
+    nodetool.flush(cql, f"{keyspace}.{table}")
+
+    return table, schema
+
+
+def simple_clustering_table(cql, keyspace):
+    table = util.unique_name()
+    schema = f"CREATE TABLE {keyspace}.{table} (pk int, ck int, v int, PRIMARY KEY (pk, ck))"
+
+    cql.execute(schema)
+
+    for pk in range(0, 10):
+        for ck in range(0, 10):
+            cql.execute(f"INSERT INTO {keyspace}.{table} (pk, ck, v) VALUES ({pk}, {ck}, 0)")
+
+    nodetool.flush(cql, f"{keyspace}.{table}")
+
+    return table, schema
+
+
+def clustering_table_with_collection(cql, keyspace):
+    table = util.unique_name()
+    schema = f"CREATE TABLE {keyspace}.{table} (pk int, ck int, v map<int, text>, PRIMARY KEY (pk, ck))"
+
+    cql.execute(schema)
+
+    for pk in range(0, 10):
+        for ck in range(0, 10):
+            map_vals = {f"{p}: '{c}'" for p in range(0, pk) for c in range(0, ck)}
+            map_str = ", ".join(map_vals)
+            cql.execute(f"INSERT INTO {keyspace}.{table} (pk, ck, v) VALUES ({pk}, {ck}, {{{map_str}}})")
+
+    nodetool.flush(cql, f"{keyspace}.{table}")
+
+    return table, schema
+
+
+def clustering_table_with_udt(cql, keyspace):
+    table = util.unique_name()
+    create_type_schema = f"CREATE TYPE {keyspace}.type1 (f1 int, f2 text)"
+    create_table_schema = f" CREATE TABLE {keyspace}.{table} (pk int, ck int, v type1, PRIMARY KEY (pk, ck))"
+
+    cql.execute(create_type_schema)
+    cql.execute(create_table_schema)
+
+    for pk in range(0, 10):
+        for ck in range(0, 10):
+            cql.execute(f"INSERT INTO {keyspace}.{table} (pk, ck, v) VALUES ({pk}, {ck}, {{f1: 100, f2: 'asd'}})")
+
+    nodetool.flush(cql, f"{keyspace}.{table}")
+
+    return table, "; ".join((create_type_schema, create_table_schema))
+
+
+def table_with_counters(cql, keyspace):
+    table = util.unique_name()
+    schema = f"CREATE TABLE {keyspace}.{table} (pk int PRIMARY KEY, v counter)"
+
+    cql.execute(schema)
+
+    for pk in range(0, 10):
+        for c in range(0, 4):
+            cql.execute(f"UPDATE {keyspace}.{table} SET v = v + 1 WHERE pk = {pk};")
+
+    nodetool.flush(cql, f"{keyspace}.{table}")
+
+    return table, schema
+
+
+@pytest.fixture(scope="module", params=[
+        simple_no_clustering_table,
+        simple_clustering_table,
+        clustering_table_with_collection,
+        clustering_table_with_udt,
+        table_with_counters,
+])
+def scylla_sstable(request, tmp_path_factory, cql, test_keyspace, scylla_only):
+    workdir = request.config.getoption('workdir')
+    scylla_path = request.config.getoption('scylla_path')
```

Comment:
We need these tests to be skipped - rather than fail with weird errors, if running against Cassandra (i.e., test/cql-pytest/run-cassandra) or against some user-installed Scylla (run Scylla yourself and then 'cd test/cql-pytest; run --host ..').
So I think you need to check if workdir or scylla_path aren't set, and if they aren't, use `pytest.skip()`. You can see examples of this in other tests that rely on Scylla-only features.

Reply:
Oh, I missed the scylla_only. So it solves the test/cql-pytest/run-cassandra problem, but it's not enough:

As I noted some users like to run Scylla on their own, and then do `cd test/cql-pytest; pytest'. The script `test/cql-pytest/run` is a very convenient script, but you don't **have** to use it. Other users want to run Scylla via docker, or even on some other machine (e.g., Scylla cloud) and tell pytest to test it. In all these cases "scylla_only" will detect that Scylla is being tested (it's not Cassandra), but it doesn't mean we have workdir and scylla_path.

You can easily check here if workdir/scylla-path is null, and if it is, pytest.skip.